gh-actions/common/build-debian/action.yml

```diff
@@ -160,6 +164,40 @@ runs:
         path: ${{ env.SOURCE_OUTPUT_DIR }}/
         if-no-files-found: error
 
+    - name: Run lintian on sources
+      if: inputs.lintian != 'false'
+      uses: kohlerdominik/docker-run-action@v2.0.0
+      with:
+        image: ${{ inputs.docker-image }}
+        environment:
+          TERM=dumb
+          DEBIAN_FRONTEND=noninteractive
+          SOURCE_OUTPUT_DIR=${{ env.SOURCE_OUTPUT_DIR }}
+        volumes: ${{ env.SOURCE_OUTPUT_DIR }}:${{ env.SOURCE_OUTPUT_DIR }}
+        workdir: ${{ env.SOURCE_OUTPUT_DIR }}
+        shell: bash
+        run: |
+          set -eu
+
+          echo 'APT::Get::Assume-Yes "true";' > /etc/apt/apt.conf.d/90aptyes
+          apt update
+          apt dist-upgrade
+
+          echo "::group::Install lintian"
+          apt install adduser lintian
+          echo "::endgroup::"
+
+          echo "::group::Create test user"
+          adduser --disabled-password --gecos "" tester
+          chown tester:tester . -R
+          echo "::endgroup::"
+
+          echo "::group::Run lintian"
+          runuser -u tester -- \
+            lintian --pedantic --fail-on error ${{ inputs.lintian }} -- \
+              "${{ env.PKG_NAME }}_${{ env.PKG_VERSION }}_source".changes
```

Comment:
Do you think we can annotate the Github Action CI build itself with the debug:warnings/error from it? If you want to see how to write something there, you can use something like https://github.com/canonical/desktop-engineering/blob/main/gh-actions/common/print-summary/action.yml#L33. Or you can format special warnings to be printed.

Reply:
Yeah, you're reading my mind on that.... I was in fact actually working on a script to generate that right now :-D